### PR TITLE
feat(build): add typescript support in build

### PR DESCRIPTION
**Pull request by niekvanstaveren**
## Motivation

Adding the support for typescript in nebula build

### CODE REVIEW

File: commands/build/lib/build.js [hunk 1]
```diff
@@ -54,6 +55,7 @@ const config = ({
 
   const auth = typeof author === 'object' ? `${author.name} <${author.email}>` : author || '';
   const moduleName = name.split('/').reverse()[0];
+  const extensions = ['.mjs', '.js', '.jsx', '.json', '.node', '.tsx', '.ts'];
```
Comment (Caele) [inline commands/build/lib/build.js:58]: Only add .ts and .tsx when typescript is enabled?

File: commands/build/lib/init-config.js [hunk 2]
```diff
@@ -33,6 +33,11 @@ const options = {
     type: 'string',
     default: undefined,
   },
+  typescriptEnabled: {
```
Comment (Caele) [inline commands/build/lib/init-config.js:36]: just name it `typescript`

File: commands/build/lib/build.js (hunk 1 above)
Comment (niekvanstaveren) [reply commands/build/lib/build.js:58]: Yes that is a good idea

**Review by T-Wizard — APPROVED** [review]

**Review by veinfors — APPROVED** [review]

File: commands/build/lib/build.js [hunk 3]
```diff
@@ -83,13 +88,14 @@ const config = ({
...
         nodeResolve({
-          extensions: ['.mjs', '.js', '.jsx', '.json', '.node'],
+          extensions,
         }),
         commonjs(),
         json(),
         babel({
           babelrc: false,
           inputSourceMap: false, // without this you get wrong source maps, but I don't know why
+          extensions,
```
Comment (Caele) [inline commands/build/lib/build.js:98]: this is a bit odd, but as long as it works.

**Review by Caele — APPROVED** [review]

**Review by LiKang6688 — APPROVED** [review]
Look forward to using it in the real case, could start with sn-pivot-table

**Pull request merged**